### ISSUE TITLE
Accessing TestCaseAttribute properties

### ISSUE THREAD

**Issue by LiohAu** [nunit/nunitv2#42]
Is it possible to retrieve the values of TestCaseAttribute programmatically (like the Description?) ? 

I could obtain the TestCategoryAttribute from TestContext.CurrentContext.Test.Properties, but I did not find anything about the TestCaseAttribute.

I know NUnit v2 is not supported anymore, but maybe you are still reading questions ?

PS : At the moment I cannot migrate to NUnit v3 (because Xamarin Test Cloud does not support it at this time).

**Comment by CharliePoole** [comment]
The properties of a test should include any specified categories without regard to how the categories were specified. However, there's a chance that the named property will be "Categories" in one case and "_CATEGORIES" in the other, depending on what version you are running.

**Comment by LiohAu** [comment]
It does include the categories, but it does not include the TestCaseAttribute.

I don't know if that was clear, so here is an example of the thing I want to obtain : 

```
[NUnit.Framework.TestCaseAttribute("...", Description="*Some description that I would like to retrieve*")]
```

**Comment by CharliePoole** [comment]
Attributes are not listed in TestContext anywhere. You can obtain properties, not attributes, from TestContext. Of course, where attributes set properties, you can get those. 

Take a look at the XML output for a test run. It will show you exactly what properties are available on each test.

**Comment by LiohAu** [comment]
If I find time to make a patch in NUnit and a PR on that repository, is there a way to get a 2.6.5 ?

**Comment by rprouse** [comment]
@LiohAu, our plan is to only release a new 2.6 version for critical issues, not for enhancements. If this feature is a priority for you, then a private fork might be the solution.

**Comment by LiohAu** [comment]
It looks like I have all the details I need in `NUnit.Core.TestExecutionContext.CurrentContext.CurrentTest.Tests` except that I still did not find how to determine which test I am currently on, is it somewhere ? even in a private var in a runner somewhere (it's not an issue if I need to use a reflection based hack)

**Comment by CharliePoole** [comment]
The `CurrentTest` property **is** the test you are currently on. Its
`Tests` property would be child tests and should be empty in the case of an
individual test case.

As I stated in my earlier post. I'm convinced that the info you want is
already available to you without modifying NUnit. I suggested what info you
need to provide us in order for us to help you, but you haven't responded.

2016-09-14 13:56 GMT-07:00 Julien notifications@github.com:
